File: FRONT-STAGE2/interfaces/QueryResponseDTO.ts
import DataFilterDTO from './DataFilterDTO';

interface QueryResponseDTO {
  id: string;
  queryName: string;
  username: string;
  description: string;
  dataFilter: DataFilterDTO;
  comments: Comment[];
}

interface Comment {
  id: string;
  comment: string;
  username: string;
  query: string;
}

export default QueryResponseDTO;

File: BACK-STAGE2/src/main/java/com/jacobo0312/backstage2/dto/QueryResponseDTO.java
package com.jacobo0312.backstage2.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class QueryResponseDTO {

    private String id;
    private String queryName;
    private String username;
    private String description;
    private List<CommentResponseDTO> comments;
    private DataFilterDTO dataFilter;
}
